### PR TITLE
[Refactor] improved UI test to use new acs token every time joining a call instead of reusing the same asc token

### PR DESCRIPTION
**Pull request by kevinyulu**
## Purpose

* Improved UI test to use new acs token every time joining a call instead of reusing the same asc token

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* Run the XCUItest to see if all tests are passed.

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/Views/SettingsView.swift [hunk 1]
```diff
@@ -52,6 +52,15 @@ struct SettingsView: View {
         Section(header: Text("Local Participant Settings")) {
             Toggle("Use expired token", isOn: $envConfigSubject.useExpiredToken)
                 .accessibilityIdentifier(AccessibilityId.expiredAcsTokenToggleAccessibilityID.rawValue)
+            Button("Generate a new token") {
+                let url = URL(string: envConfigSubject.acsTokenUrl)!
```
Comment (AmyL219) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/Views/SettingsView.swift:56]: Why do we want to have the acsTokenUrl? Which acsTokenUrl we would like to use though? 
The acsTokenUrl we want to use in the E2E test need to be secured. A self host token url  ( like `https://xxxxx.azurewebsites.net/api/Auth?code=...` without authentication is not allowed.

Comment (kevinyulu) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/Views/SettingsView.swift:56]: Would we use this acsTokenUrl in the E2E test?
Yes it's used in some UI tests:

```
testCallCompositeEndCallGroupCallSwiftUI()
testCallCompositeEndCallTeamsCallSwiftUI()
testCallCompositeEndCallGroupCallUIKit()
testCallCompositeEndCallTeamsCallUIKit()
```
Which acsTokenUrl we would like to use though?
I think currently I test it with a self host one in my local test. we can use a secured one in the info.plist from app center / local. it should work.

Comment (palatter) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/Views/SettingsView.swift:56]: So if we use a secured url, which user is used when running the pipeline?

Comment (AmyL219) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/Views/SettingsView.swift:56]: Secured one would need user to login to get the access token. 
The pipeline has one step to fetch the token every run inside. 
See this step `Get access token in pip3` in this pipeline: https://skype.visualstudio.com/SCC/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=12410

**Comment by kevinyulu** [comment]
closed pull request as the the team has not made a decision of the approach to use acs token url in the ui test

**Pull request closed without merging**